examples/model_selection/plot_train_error_vs_test_error.py

```diff
@@ -73,7 +73,7 @@
     linewidth=3,
     label="Optimum on test",
 )
-plt.legend(loc="lower left")
+plt.legend(loc="center left")
```

Comment:
The updated figure looks like this:

![Screen Shot 2022-08-10 at 3 46 06 PM](https://user-images.githubusercontent.com/5402633/184006529-193aeee6-84a8-4ca1-ba1f-6a771f9e03aa.jpg)

To me, it still covers the lines too much. What do you think about placing the legend on the bottom right?